tests/integration/targets/cassandra_role/tasks/main.yml

```diff
@@ -117,6 +140,26 @@
     that:
       - "'app_user' in myrole.stdout"
 
+- name: Create a role (check mode) - non-SSL cluster
+  community.cassandra.cassandra_role:
+    name: no_ssl_app_user
+    password: 'secretZHB78'
+    state: present
+    login: yes
+    ssl: False
+    login_user: "{{ cassandra_admin_user }}"
+    login_password: "{{ cassandra_admin_pwd }}"
+  check_mode: yes
+
+- name: Get output of list roles
+  ansible.builtin.shell: cqlsh --username "{{ cassandra_admin_user }}" --password "{{ cassandra_admin_pwd }}" --execute "LIST ROLES"
+  register: nonssl_myrole
+
+- name: Assert no_ssl_app_user exists in output
```

Comment:
You're using check_mode: yes above so this role shouldn't exist and this assert will always fail.

Reply:
Yep, fixed in https://github.com/ansible-collections/community.cassandra/pull/230/commits/5ef4e7b36719cd8b323b10445189338685016e2a